pkg/cli/image/info/info.go

```diff
@@ -87,35 +91,67 @@ type InfoOptions struct {
 	SecurityOptions imagemanifest.SecurityOptions
 	FilterOptions   imagemanifest.FilterOptions
 
-	Images []string
-
-	FileDir string
+	operatorClient operatorv1alpha1client.OperatorV1alpha1Interface
 
-	Output string
+	Images   []string
+	FileDir  string
+	Output   string
+	ICSPFile string
 }
 
-func (o *InfoOptions) Complete(cmd *cobra.Command, args []string) error {
+func (o *InfoOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
 	if len(args) < 1 {
 		return fmt.Errorf("info expects at least one argument, an image pull spec")
 	}
 	o.Images = args
-	return nil
-}
 
-func (o *InfoOptions) Validate() error {
-	return o.FilterOptions.Validate()
+	clientConfig, err := f.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	o.operatorClient, err = operatorv1alpha1client.NewForConfig(clientConfig)
+	if err != nil {
+		return err
+	}
+
+	return nil
 }
 
-func (o *InfoOptions) Run() error {
+func (o *InfoOptions) Validate(cmd *cobra.Command) error {
 	if len(o.Images) == 0 {
 		return fmt.Errorf("must specify one or more images as arguments")
 	}
+	if len(o.ICSPFile) > 0 {
+		warned := false
+		for _, location := range o.Images {
+			if warned {
+				break
+			}
+			sources, err := imagesource.ParseSourceReference(location, nil)
```

Comment:
You have to handle wildcard expansion here.  Do this inline in Run() (otherwise you have to lift the entire loop out of Run which is a major refactoring (and likely to break other stuff).